File: prob53/main.py
#!/usr/bin/env python3

from math import factorial

def c(n,r):
    return factorial(n) / factorial(r) / factorial(n - r)

i = 0; lowr = 1; highr = 100; limit = 1000000
for n in range(lowr, highr + 1):
    for r in range(lowr, n + 1):
        if c(n,r) > limit:
            i += 1

print('For', lowr, '<= n <=', highr, ', the number of values of C(n, r) that exceed', limit, 'is:', i)

File: prob53/main.cpp
#include <iostream>

using namespace std;

int main() {
    const unsigned int high = 100, limit = 1000000;
    unsigned int sum = 0, c[high + 2][high + 2];

    c[1][1] = 1;
    for (unsigned int row = 2; row < high + 2; ++row) {
       c[row][row] = 1;
       for (unsigned int column = row - 1; column > 1; --column) {
           c[row][column] = c[row - 1][column - 1] + c[row - 1][column];
           if (c[row][column] > limit) {
               ++sum;
           }
        }
        c[row][1] = 1;
    }
    sum += 1 > limit ? (2 * high + 1) : 0;
    //cout << c[23][11] << endl;

    cout << "For " << 1 << " <= n <= " << high << ", the number of values of C(n, r)"
         << " that exceed " << limit << " is: " << sum << endl;
}
